### PR TITLE
LEAF-2452 - copy existing record by

### PR DESCRIPTION
**Pull request by shaneodd**
On a record, there is now a new button under tools called Duplicate From. This will have a popup that will allow a user to choose a new title, service, adjust priority and choose which sections of the record to be copied over.

### CODE REVIEW

File: LEAF_Request_Portal/templates/print_form.tpl [hunk 1]
```diff
@@ -461,42 +484,296 @@ function cancelRequest() {
...
         });
     });
 }
 
+/**
+ *
+ * @param {object} indicators
+ * @returns {array}
+ */
+function getChildrenIndicatorIDs(indicators) {
```
Comment (Pelentan) [inline LEAF_Request_Portal/templates/print_form.tpl:519]: You check to see if it's null, but you don't verify that it's an object with the needed key-values.  How will javascript react if "indicators" is "plaid"?

File: LEAF_Request_Portal/templates/print_form.tpl [hunk 2]
```diff
@@ -461,42 +484,296 @@ function cancelRequest() {
...
+                success: function (res) {
+                    Object.values(res).forEach(function (resultValue) {
+
+                        if (pickAndChooseValues.includes(resultValue[series].indicatorID)) {
+
+                            // uploaded files will need to have a special case done to them to copy them over to the new record
+                            if(resultValue[series].format == 'fileupload' || resultValue[series].format == 'image'){
+                                if(resultValue[series].value.length > 0){
+                                    resultValue[series].value.forEach(function(currentFile){
+                                        var fileDat = {
```
Comment (Pelentan) [inline LEAF_Request_Portal/templates/print_form.tpl:669]: Why is this a var instead of let?  Looks like it is only used here.

**Review by Pelentan — CHANGES_REQUESTED** [review]

**Review by jampaul3 — APPROVED** [review]

File: LEAF_Request_Portal/templates/print_form.tpl (hunk 2 above)
Comment (shaneodd) [reply LEAF_Request_Portal/templates/print_form.tpl]: This was something I missed, was testing something and after I realized what I was doing I forgot to change it back.

**Review by aerinkayne — APPROVED** [review]

**Review by pete-nerantzinis — APPROVED** [review]

**Review by Pelentan — APPROVED** [review]

File: LEAF_Request_Portal/templates/print_form.tpl [hunk 3]
```diff
@@ -461,42 +484,302 @@ function cancelRequest() {
...
+
+            // make sure indicatorID exists
+            if(indicator.indicatorID !== undefined){
+                children.push(indicator.indicatorID);
+            }
+
+            // make sure child exists
+            if (indicator.child !== undefined) {
+                let subchildren = getChildrenIndicatorIDs(indicator.child);
+                // well this took me a bit to realize concat returns the value of results
```
Comment (raphaelfretter) [inline LEAF_Request_Portal/templates/print_form.tpl:533]: I don't know if this is a necessary comment lol

**Review by raphaelfretter — CHANGES_REQUESTED** [review]

**Review by AmandaSumner — APPROVED** [review]

**Review by maxrichardjr — APPROVED** [review]

**Pull request merged**